packages/camera/camera/example/lib/main.dart

```diff
@@ -651,39 +651,44 @@ class _CameraExampleHomeState extends State<CameraExampleHome>
       imageFormatGroup: ImageFormatGroup.jpeg,
     );
 
-    controller = cameraController;
-
     // If the controller is updated then update the UI.
     cameraController.addListener(() {
       if (mounted) {
         setState(() {});
       }
       if (cameraController.value.hasError) {
         showInSnackBar(
-            'Camera error ${cameraController.value.errorDescription}');
+          'Camera error ${cameraController.value.errorDescription}',
+        );
       }
     });
 
     try {
       await cameraController.initialize();
+      if (cameraController.value.hasError) {
+        showInSnackBar(
+          'Camera error ${cameraController.value.errorDescription}',
+        );
+        return;
+      }
       await Future.wait(<Future<Object?>>[
         // The exposure mode is currently not supported on the web.
-        ...!kIsWeb
-            ? <Future<Object?>>[
-                cameraController.getMinExposureOffset().then(
-                    (double value) => _minAvailableExposureOffset = value),
-                cameraController
-                    .getMaxExposureOffset()
-                    .then((double value) => _maxAvailableExposureOffset = value)
-              ]
-            : <Future<Object?>>[],
+        if (kIsWeb) ...<Future<Object?>>[
+          cameraController
+              .getMinExposureOffset()
+              .then((double value) => _minAvailableExposureOffset = value),
+          cameraController
+              .getMaxExposureOffset()
+              .then((double value) => _maxAvailableExposureOffset = value)
+        ],
         cameraController
             .getMaxZoomLevel()
             .then((double value) => _maxAvailableZoom = value),
         cameraController
             .getMinZoomLevel()
             .then((double value) => _minAvailableZoom = value),
       ]);
+      controller = cameraController;
```

Comment:
i'm not too sure why we are moving down this line. Can you explain more?

Reply:
I've added some comments here:
```
// `controller` needs to be set until it's fully initialized, to avoid a
// race condition when the lifecycle callback will initialize a new one,
// causing a new `controller` to be created. This happens when the camera
// permission dialog dismisses at the first run, which triggers
// `didChangeAppLifecycleState` that disposes and re-creates the controller.
```

Reply:
> to avoid a race condition when the lifecycle callback being called and creates a new `controller`. 

Can you explain what's bad about this? does it crash? How?

Reply:
> Can you explain what's bad about this?

Will the comparison https://github.com/flutter/plugins/pull/5748#issue-1236204460 explains this?

> does it crash? How?

It won't crash, but switching controllers in a short circumstance might cause the preview flashes or bad state management. Since this is the example of a guideline for using the camera, we might always provide a safe/better way to use controllers.

Reply:
Gotcha. Make sense. Could you add this to the comment?